actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -53,10 +37,13 @@ def expanded_cache_key(key)
         key.frozen? ? key.dup : key # #read_multi & #write may require mutability, Dalli 2.6.0.
```

Comment:
Check out the explicit `@template` use just above to extract a `virtual_path` because we reference this, we can't support collections rendering multiple templates. Well, we can, there's just no telling that the collection cache reuses the individual fragment caching.

I don't know of a better way to get a hold of the `virtual_path` from here than accessing it through the templates. This likely means we'll have to split switch to eager-like loading of the templates as done here: https://github.com/rails/rails/pull/19577/files#diff-738cc3c10058587e78559e48066ad0bfR58 (another stab at supporting multiple templates).